ac/ac-dual/src/ActivityRunner.jsx

```diff
@@ -0,0 +1,235 @@
+    wait: 'Waiting for next question',
+    end: 'Activity completed! Thank you'
+  },
+  fr: {
+    start: 'Commencer',
+    yes: 'OUI',
+    no: 'NON',
+    question: 'Sont-ils symétriques?',
+    wait: 'Attendez la question suivante',
+    end: 'Activité terminée! Merci'
+  }
+};
+
+let noAnswerTimeout;
+let delayTimeout;
+
+const Form = withState('language', 'setLanguage', null)(
+  ({ language, setLanguage, onSubmit, name }) => (
+    <React.Fragment>
+      <div style={styles.text}>Welcome {name}!</div>
+      <div style={styles.text}>Choisis ton langage</div>
+      <div style={styles.text}>Choose your language</div>
+      <div style={styles.commands}>
+        <Button
+          style={{ ...styles.button, left: 0 }}
+          bsStyle={language === 'fr' ? 'success' : 'default'}
+          onClick={() => setLanguage('fr')}
+        >
+          Français
+        </Button>
+        <Button
+          style={{ ...styles.button, right: 0 }}
+          bsStyle={language === 'en' ? 'success' : 'default'}
+          onClick={() => setLanguage('en')}
+        >
+          English
+        </Button>
+      </div>
+      {language !== null && (
+        <div style={{ ...styles.commands, width: '100px' }}>
+          <Button
+            style={{ ...styles.button, width: '100%' }}
+            onClick={() => onSubmit(language)}
+            bsStyle="primary"
+          >
+            Submit
+          </Button>
+        </div>
+      )}
+    </React.Fragment>
+  )
+);
+
+const Guidelines = ({ start, guidelines, lang }) => (
+  <React.Fragment>
+    <div style={styles.guidelines}>{guidelines}</div>
+    <div style={{ ...styles.commands, width: '120px' }}>
+      <Button style={{ ...styles.button, width: '100%' }} onClick={start}>
+        {texts[lang].start}
+      </Button>
+    </div>
+  </React.Fragment>
+);
+
+const CountDownTimer = TimedComponent(({ timeNow, length, start }) => {
+  const timeLeft = Math.ceil((length - Math.ceil(timeNow - start)) / 1000);
+  return <div style={styles.text}>{timeLeft + ' s'}</div>;
+}, 100);
+
+const Delay = ({ next, delay, lang }) => {
+  clearTimeout(delayTimeout);
+  delayTimeout = setTimeout(next, delay);
+  return (
+    <React.Fragment>
+      <div style={styles.text}>{texts[lang].wait}</div>
+      <CountDownTimer start={Date.now()} length={delay} />
+    </React.Fragment>
+  );
+};
+
+const randSymmetry = figure => {
+  figure.symmetrical = Math.random() < 0.5;
+  return figure;
+};
+
+const Question = props => {
+  const { setQuestion, startTime, logger, data, dataFn, activityData } = props;
+  const lang = data.language;
+
+  const onClick = answer => () => {
+    clearTimeout(noAnswerTimeout);
+    // Logs the question and answer provided
+    const answerTime = Date.now();
+    // Increases the progress and logs the new progress
+    dataFn.numIncr(1, 'progress');
+    // Increases the score and logs the new score
+    const isCorrectAnswer = FIGURE.symmetrical === answer ? 1 : 0;
```

Comment:
Wouldn't it make more sense to have this `FIGURE` object be in the state or props of the `Question` component rather than a global variable? What do you think?